utils/timer/executor.go

```diff
@@ -9,11 +9,12 @@ import (
 
 // Executor ...
 type Executor struct {
```

Comment:
It's not clear to me what the changes in this file are meant to do. What's the purpose of this change?

Reply:
It's meant to separate the lock held during dispatch and the lock held to add an event. So these changes are meant to make it so that calling Add won't block even if a function being executed within the Dispatch goroutine is currently blocking.

Reply:
Calling Add doesn't block when a function is being executed in the current code.

Reply:
You're right, just reverted the changes